sentry/src/main/java/io/sentry/Baggage.java

```diff
@@ -217,10 +218,12 @@ public Baggage(
       final @Nullable Double sampleRate,
       final @Nullable Double sampleRand,
       final @Nullable String thirdPartyHeader,
-      boolean isMutable,
-      boolean shouldFreeze,
+      final boolean isMutable,
+      final boolean shouldFreeze,
       final @NotNull ILogger logger) {
-    this.keyValues = keyValues;
+    // TODO should we deep-copy the keyValues here?
+    // if so we could optimize this by only synchronizing in case isMutable is true
+    this.keyValues = Collections.synchronizedMap(keyValues);
```

Comment:
Instead of wrapping here, which will cause problems with virtual threads again due to using `synchronized` internally, could we revert this line and instead always use `ConcurrentHashMap` in place of `Map` in this class?